App/Species/KineticSpecies.lua

```diff
@@ -700,6 +720,19 @@ function KineticSpecies:applyBcIdx(tCurr, field, externalField, inIdx, outIdx, i
   if self.positivity then
      self:checkPositivity(tCurr, outIdx)
   end
+  if self.nonconPositivity then
+     self.numDensityCalc:advance(tCurr, {self:rkStepperFields()[outIdx]}, {self.prePosM0})
+     
+     self.nonconPos:advance(tCurr, {self:rkStepperFields()[outIdx]}, {self:rkStepperFields()[outIdx]}, false)
+     
+     self.numDensityCalc:advance(tCurr, {self:rkStepperFields()[outIdx]}, {self.postPosM0})
+     self.delPosM0:combine(1.0, self.postPosM0, -1.0, self.prePosM0)
+
+     local tm, lv = self.intPosM0:lastData()
+     self.calcIntPosM0:advance(tCurr, {self.delPosM0}, {self.intDelPosM0})
+     local tmDel, lvDel = self.intDelPosM0:lastData()
+     self.intPosM0:appendData(tmDel, {lv[1]+lvDel[1]})
+  end
```

Comment:
I think we shouldn't add more branching in code that gets called inside the time loop. So I would advocate for/request that the code instead checks the positivity booleans and creates appropriate functions during initialization. When the booleans are false those functions would be empty. See App/Species/FluidSpecies.lua, which is probably the most advanced in this regard. Its :applyBcIdx method simply calls a function self.checkPositivity whose definition depends on the booleans in the input file. If `positivity` and `nonconPositivity` are exclusive, then you could create a single function that does either.
The hope is that such approach reduces overhead. The compiler either simply inlines the function or in the case of empty functions, removes it altogether.

Reply:
Sounds good. Will do.

Reply:
Commit c7b8dcd addresses this.